packages/aws-cdk/lib/api/cloudformation-deployments.ts

```diff
@@ -366,6 +360,115 @@ export class CloudFormationDeployments {
     return stack.exists;
   }
 
+  private async readCurrentNestedTemplate(stackArtifact: cxapi.CloudFormationStackArtifact, nestedStackName: string): Promise<Template> {
+    return this.readCurrentStackTemplate(stackArtifact, nestedStackName);
+  }
+
+  private async replaceNestedStacksInRootStack(rootStackArtifact: cxapi.CloudFormationStackArtifact, deployedTemplate: any) {
+    if (!this.stackHasNestedStacks(rootStackArtifact)) {
+      return;
+    }
+
+    const sdkWithRole = await prepareSdkWithLookupRoleFor(this.sdkProvider, rootStackArtifact);
+    const listStackResourcesByStackName: { [key:string] : ListStackResources } = {};
+    await this.replaceNestedStacksInParentTemplate(
+      rootStackArtifact, rootStackArtifact.template, deployedTemplate, rootStackArtifact.stackName, listStackResourcesByStackName, sdkWithRole.sdk,
+    );
+  }
+
+  private stackHasNestedStacks(stackArtifact: cxapi.CloudFormationStackArtifact): boolean {
+    return Object.values(stackArtifact.template.Resources ?? {}).some((resource: any) => resource.Type === 'AWS::CloudFormation::Stack');
+  }
+
+  private async replaceNestedStacksInParentTemplate(
+    rootStackArtifact: cxapi.CloudFormationStackArtifact, generatedParentTemplate: any, deployedParentTemplate: any,
+    parentStackName: string | undefined, listStackResourcesByStackName: { [key: string] : ListStackResources }, sdk: ISDK,
+  ) {
+    const idResourceMap = Object.entries(generatedParentTemplate.Resources ?? {}).filter(([_, attrs]) => this.isNestedStackResource(attrs));
+    for (const [nestedStackLogicalId, resource] of idResourceMap) {
+      let nestedStackResource = resource as any;
+
+      if (parentStackName && !listStackResourcesByStackName[parentStackName]) {
+        listStackResourcesByStackName[parentStackName] = new LazyListStackResources(sdk, parentStackName);
+      }
+
+      const assetPath = nestedStackResource.Metadata['aws:asset:path'];
+      const nestedStackTemplates = await this.getNestedStackTemplates(
+        rootStackArtifact, assetPath, nestedStackLogicalId, parentStackName, listStackResourcesByStackName,
+      );
+
+      generatedParentTemplate.Resources[nestedStackLogicalId] = nestedStackTemplates.generatedNestedTemplate;
+      generatedParentTemplate.Resources[nestedStackLogicalId].Type = 'AWS::CloudFormation::Stack';
+
+      if (deployedParentTemplate.Resources) {
+        deployedParentTemplate.Resources[nestedStackLogicalId] = nestedStackTemplates.deployedNestedStackTemplate;
+        deployedParentTemplate.Resources[nestedStackLogicalId].Type = 'AWS::CloudFormation::Stack';
+      }
```

Comment:
Also, doesn't the Stack resource have properties of its own, like `NotificationArns`, `Parameters`, `Tags`, etc.?

If you override the entire resource with the template, you will erase all of these properties of the `AWS::CloudFormation::Stack` resource from the diff output, which cannot be the correct thing to do.